FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckErrorUtil.m

```diff
@@ -35,14 +47,21 @@ + (NSError *)cachedTokenExpired {
                      underlyingError:nil];
 }
 
++ (NSError *)keychainErrorWithError:(NSError *)error {
+  NSString *failureReason = [NSString stringWithFormat:@"Keychain access error."];
+  return [self appCheckErrorWithCode:FIRAppCheckErrorCodeKeychain
```

Comment:
optional: What if `error` passed is not keychain one? I would check the expected domain/codes and fallback to `FIRAppCheckErrorCodeUnknown`.

Reply:
I moved some things around and now check the domain/codes, falling back to `FIRAppCheckErrorCodeUnknown` if necessary. PTAL